src/Core/Compiler/Extensions/TypeSymbolExtensions.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using ScriptSharp.ScriptModel;
+
+namespace ScriptSharp.Extensions
+{
+    public static class TypeSymbolExtensions
+    {
+        internal static ICollection<InterfaceSymbol> GetInterfaces(this TypeSymbol symbol)
+        {
+            if (symbol is ClassSymbol classSymbol)
+            {
+                return classSymbol.Interfaces ?? new InterfaceSymbol[0];
+            }
+
+            if (symbol is InterfaceSymbol interfaceSymbol)
+            {
+                var interfaces = new List<InterfaceSymbol> { interfaceSymbol };
+
+                if (interfaceSymbol.Interfaces != null)
+                {
+                    interfaces.AddRange(interfaceSymbol.Interfaces);
+                }
+
+                return interfaces;
+            }
+
+            return null;
+        }
+
+        internal static bool IsCollectionType(this TypeSymbol symbol)
```

Comment:
We could shorten this method with null chaining. 
```csharp
internal static bool IsCollectionType(this TypeSymbol symbol)
{
    return symbol?.GetInterfaces()
        ?.Any(i => i.FullName.StartsWith("System.Collections") && i.FullGeneratedName == "ICollection")
        ?? false;
}
```

Reply:
I find my version easier to maintain (add more logic) and debug, but up to you

Reply:
Honestly i don't mind at the end of the day :D